tool/tsh/config.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"text/template"
+
+	"github.com/gravitational/teleport/api/profile"
+	"github.com/gravitational/teleport/api/utils/keypaths"
+	"github.com/gravitational/trace"
+)
+
+const sshConfigTemplate = `
+# Common flags for all {{ .clusterName }} hosts
+Host *.{{ .clusterName }} {{ .proxyHost }}
+    UserKnownHostsFile "{{ .knownHostsPath }}"
+
+# Flags for all {{ .clusterName }} hosts except the proxy
+Host *.{{ .clusterName }} !{{ .proxyHost }}
+    Port 3022
+    {{- if .leaf }}
+    ProxyCommand ssh -p {{ .proxyPort }} {{ .proxyHost }} -s proxy:$(echo %h | cut -d '.' -f 1):%p@{{ .clusterName }}	
```

Comment:
Can `$(echo %h | cut -d '.' -f 1)` be done by Go code instead and pasted here as a template variable?

Same below.

Reply:
It is possible but would require another subcommand of some sort to be built into tsh. Per advice from @russjones we've decided to defer for a bit as the upcoming `tsh connect` functionality could supersede any text manipulation we do.